### PR TITLE
fix(webapp):  Added Quiz State in persistence model to handle Latecomer participants

### LINKED ISSUE #79: Latecomers will break quiz
A late participant who has the link can join the quiz after the Quiz Master has already started the quiz session.  This will break the quiz for the Quiz Master with a TypeError: contestantsList.filter(...)[0] is undefined

Desired behaviour will be either 
a)  The participant can safely join the quiz even if it has already started 
or
b) The participant will receive a message "Sorry the quiz has already started"
or 
c)  Quiz Master can be given the choice to admit a late participant

### PR DESCRIPTION
**Pull request by richielewis**
Quiz DB object now includes QuestionNo and QuizState which are updated as the Quiz Master moves the Quiz along. 

If a participant tries to join a quiz which has already started they will received a sorry message and be unable to join. 

Fixes #79

### CODE REVIEW

File: QuizMaster.Api/Controllers/QuizzesController.cs [hunk 1]
```diff
@@ -54,6 +54,20 @@ public async Task<ActionResult<String>> GetQuizName(string id)
...
+        {
+            var quiz = await mediator.Send(new Details.Query(id));
+
+            if (quiz == null)
+            {
+                return NotFound();
+            }
+
+            return Ok(new Details.QuizStateValues(){QuizState = quiz.State, QuestionNo = quiz.QuestionNo});
+            //mediator.Send(new Details.Query(id)).Result.Name);
```
Comment (calumtomeny) [inline QuizMaster.Api/Controllers/QuizzesController.cs:68]: Remove the commented code.

File: QuizMaster.Application/Quizzes/Update.cs [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+            public int? QuestionNo { get; set;}
+        }        
+
+        public class Handler : IRequestHandler<Command, Quiz>
+        {
+            private readonly QuizContext context;
+            public Handler(QuizContext context)
+            {
+                this.context = context;
+
```
Comment (calumtomeny) [inline QuizMaster.Application/Quizzes/Update.cs:37]: Extra space.

File: QuizMaster.Application/Quizzes/Update.cs [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+            {
+                var quiz = await context.Quiz.SingleOrDefaultAsync(x => x.Code == request.QuizCode);
+                //var save = false;
+                if(request.CommandBody.QuestionNo.HasValue){
+                    quiz.QuestionNo = request.CommandBody.QuestionNo.Value;
+                    //save = true;
+                }
+                if(request.CommandBody.QuizState.HasValue){
+                    quiz.State = request.CommandBody.QuizState.Value;
+                    //save = true;
```
Comment (calumtomeny) [inline QuizMaster.Application/Quizzes/Update.cs:50]: Commented code.

File: QuizMaster.Application/Quizzes/Update.cs [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+            public Handler(QuizContext context)
+            {
+                this.context = context;
+
+            }
+
+            public async Task<Quiz> Handle(Command request, CancellationToken cancellationToken)
+            {
+                var quiz = await context.Quiz.SingleOrDefaultAsync(x => x.Code == request.QuizCode);
+                //var save = false;
```
Comment (calumtomeny) [inline QuizMaster.Application/Quizzes/Update.cs:43]: Commented code.

File: QuizMaster.Application/Quizzes/Update.cs [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+                if(request.CommandBody.QuestionNo.HasValue){
+                    quiz.QuestionNo = request.CommandBody.QuestionNo.Value;
+                    //save = true;
```
Comment (calumtomeny) [inline QuizMaster.Application/Quizzes/Update.cs:46]: Commented code.

File: QuizMaster.Application/Quizzes/Update.cs [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+                {
+                    var success = await context.SaveChangesAsync() > 0;
+                    if (success)
+                    {
+                        return quiz;
+                    }
+                }
+                else
+                {
+
```
Comment (calumtomeny) [inline QuizMaster.Application/Quizzes/Update.cs:62]: Remove extra line.

File: QuizMaster.Api/Controllers/QuizzesController.cs (hunk 1 above)
Comment (richielewis) [reply QuizMaster.Api/Controllers/QuizzesController.cs]: Done

File: QuizMaster.Application/Quizzes/Update.cs (hunk 2 above)
Comment (richielewis) [reply QuizMaster.Application/Quizzes/Update.cs]: Done

File: QuizMaster.Application/Quizzes/Update.cs (hunk 3 above)
Comment (richielewis) [reply QuizMaster.Application/Quizzes/Update.cs]: Done

File: QuizMaster.Application/Quizzes/Update.cs (hunk 4 above)
Comment (richielewis) [reply QuizMaster.Application/Quizzes/Update.cs]: Done

File: QuizMaster.Application/Quizzes/Update.cs (hunk 5 above)
Comment (richielewis) [reply QuizMaster.Application/Quizzes/Update.cs]: Done

File: QuizMaster.Application/Quizzes/Update.cs (hunk 6 above)
Comment (richielewis) [reply QuizMaster.Application/Quizzes/Update.cs]: Done

**Review by calumtomeny — APPROVED** [review]

**Pull request merged**